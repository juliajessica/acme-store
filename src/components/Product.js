import React from 'react';
import { number, string } from 'prop-types';
import Button from './Button';
import { images } from '../utils/imageArray';

import './product.scss';

const Product = ({ price, inventory, title, quantity }) => (
  <div className="product_container">
    <div className="product_details">
      <div className="product_image">
        {images.map(image => {
          return title === image.title ? (
            <img src={image.src} alt={image.alt} title={image.title} key={image.title} />
          ) : null;
        })}
      </div>
      <div className="product_details-flex">
        <span>{title}</span>
        <span>{`$${price}`}</span>
        <button type="button" className="product_remove-button">
          Remove
        </button>
      </div>
    </div>
    <div className="product_total-quantity">
      <Button
        // onCheckoutClicked={() => checkout(products)}
        theme="secondary-minus"
        disabled={quantity > 0 ? '' : 'disabled'}
      />
      <div className="product_quantity">{quantity}</div>
      <Button
        // onCheckoutClicked={() => checkout(products)}
        theme="secondary-add"
        disabled={quantity > 0 ? '' : 'disabled'}
      />
    </div>

    <p>{inventory ? `${inventory} remaining` : null}</p>
  </div>
);

Product.propTypes = {
  price: number.isRequired,
  inventory: number.isRequired,
  title: string.isRequired,
  quantity: number.isRequired
};

export default Product;
