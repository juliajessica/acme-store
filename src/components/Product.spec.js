import React from 'react';
import { shallow } from 'enzyme';
import Product from './Product';

const getDefaultProps = () => ({
  title: 'Test Product',
  price: 9.99,
  quantity: 3,
  inventory: 1,
  onAddToCart: jest.fn(),
  onRemoveFromCart: jest.fn()
});

describe('Product component', () => {
  it('should return component', () => {
    const props = getDefaultProps();
    const wrapper = shallow(<Product {...props} />);

    expect(wrapper.debug()).toMatchSnapshot();
  });
  it('should render all props', () => {
    const props = getDefaultProps();
    const wrapper = shallow(<Product {...props} />);
    console.log('quantity', wrapper.setProps());
    // console.log('check', wrapper.find('')
    // console.log('component.props()', component.props().children[1].props);
    // expect(component.props('title')).toEqual('Test Product');
    expect(wrapper.find('.product_quantity')).to.have.lengthOf(1);
  });
  // describe('when given inventory', () => {
  //   it('should render title, price, and inventory', () => {
  //     const { component } = setup({ title: 'Test Product', price: 9.99, inventory: 6 });
  //     expect(component.text()).toBe('Test Product - $9.99 x 6');
  //   });
  // });
});
