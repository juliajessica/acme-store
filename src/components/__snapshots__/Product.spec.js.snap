// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product component should return component 1`] = `
"<div className=\\"product_container\\">
  <div className=\\"product_details\\">
    <div className=\\"product_image\\" />
    <div className=\\"product_details-flex\\">
      <span>
        Test Product
      </span>
      <span>
        $9.99
      </span>
      <button type=\\"button\\" className=\\"product_remove-button\\">
        Remove
      </button>
    </div>
  </div>
  <div className=\\"product_total-quantity\\">
    <Button onClick={[Function: mockConstructor]} theme=\\"secondary-minus\\" disabled=\\"\\" customClass={{...}} type=\\"button\\" />
    <div className=\\"product_quantity\\">
      3
    </div>
    <Button onClick={[Function: mockConstructor]} theme=\\"secondary-add\\" disabled=\\"\\" customClass={{...}} type=\\"button\\" />
  </div>
  <p>
    1 remaining
  </p>
</div>"
`;
